pkg/kubelet/cm/dra/claiminfo.go

```
@@ -118,10 +148,25 @@ func (cache *claimInfoCache) hasPodReference(UID types.UID) bool {
 	defer cache.RUnlock()
 
 	for _, claimInfo := range cache.claimInfo {
-		if claimInfo.podUIDs.Has(string(UID)) {
+		if claimInfo.PodUIDs.Has(string(UID)) {
 			return true
 		}
 	}
 
 	return false
 }
+
+func (cache *claimInfoCache) syncToCheckpoint() {
+	cache.RLock()
+	defer cache.RUnlock()
+
+	claimInfoStateList := make(state.ClaimInfoStateList, 0)
+	for _, infoClaim := range cache.claimInfo {
+		claimInfoStateList = append(claimInfoStateList, infoClaim.ClaimInfoState)
+	}
+
+	err := cache.state.Store(claimInfoStateList)
+	if err != nil {
+		klog.ErrorS(err, "Failed to checkpoint claimInfo state")
```

Would it make sense to add a comment explaining why we're ignoring errors on this level?

---

Agreed, let's return the error and kill the kubelet if something goes wrong here.

---

what do you mean kill the kubelet? it will fail on PrepareResources/UnPrepareResources and kubelet will try again, right?

---

I guess we have two options -- either a `log.Fatal()` (which will kill the kubelet) or a simple error return (which will then loop indefinitely (with a backoff), failing over and over again on the checkpoint write). The second one probably makes the most sense though and is more consistent with how the rest of the manager operates.

---

Error return would be enough from my pov.

---

Sounds good